### ISSUE TITLE
igmpproxy script somehow does not do the trick

### ISSUE THREAD

**Issue by fwieringen** [coolhva/usg-kpn-ftth#50]
Hi all,

I noticed that my IPTV isn't working after a power outage. Cloud Key and USG boot up and everything is up and running. Apps on the KPN TV settop box work. Streaming earlier programs works. Watching live TV doesn't.

Logging onto the USG and performing a **igmp-proxy restart** command does the trick. Although I would expect the script to take care of this...

Any tips on debugging?

Thanks! Friso

**Comment by hapklaar** [comment]
Same here! Every time I lose power to my USG, IPTV doesn't come back up by itself again, I also have to restart using `/opt/vyatta/bin/vyatta-op-cmd-wrapper restart igmp-proxy`

**Comment by kevintuhumury** [comment]
Same here. Whenever a situation like that occurs, I also have to run `restart igmp-proxy` manually. When I do that it will always mention the following:

```
Stopping IGMP proxy.
Starting IGMP proxy service.
```

Never something else. Only once have I seen a different output. In that case I ran `pidof igmpproxy` to see whether it was running or not and it wasn't, meaning... in all other cases it is likely already running, but doesn't do the trick for some reason. 

The `igmpproxy.sh` script does exactly that when verifying whether it is running or not (`pidof igmpproxy`), if it isn't, than it will restart it and only in that case. As it's already running for me (and probably also for you guys), the script won't do anything. Even though it seems to require a restart, because after that everything works fine.

I was even thinking of removing the check and running the script every now and than, but restarting the service disrupts IPTV for a second or two, so that's not an option.

Does anyone have a fix for this?

**Comment by coolhva** [comment]
I have no way of checking if IGMP proxy is still doing its task besides checking if it is running. Maybe something can be done from the statistics but I'm not sure.

**Comment by hapklaar** [comment]
For the moment I just added `/opt/vyatta/bin/vyatta-op-cmd-wrapper restart igmp-proxy` to the end of the kpn.sh script. Seems to work. Tested with a few simulated power outages.
